static/js/components/material/Drawer.js

```diff
@@ -20,22 +23,36 @@ type DrawerProps = {
 class Drawer extends React.Component {
   drawer: null;
   drawerRoot: null;
+  collapseItemButton: null;
+  createCollectionButton: null;
   props: DrawerProps;
 
   componentDidMount() {
-    const {onDrawerClose} = this.props;
+    const { onDrawerClose, dispatch } = this.props;
     this.drawer = new MDCTemporaryDrawer(this.drawerRoot);
     this.drawer.listen('MDCTemporaryDrawer:close', onDrawerClose);
     this.updateRequirements();
 
-    // Close drawer on button click; this is a necessary hack to get around MDC limitations
-    let item = document.querySelector('#collapse_item');
-    if (item) {
-      item.addEventListener('click', (evt: MouseEvent) => {
-        evt.preventDefault();
+    // Attach click listeners here; this is a necessary hack to get around MDC limitations
+    if (this.collapseItemButton) {
+      // make flow happy
+      this.collapseItemButton.addEventListener('click', (event: MouseEvent) => {
+        event.preventDefault();
         onDrawerClose();
       }, false);
     }
+
+    // this will be undefined if SETTINGS.editable is false
+    if (this.createCollectionButton) {
+      this.createCollectionButton.addEventListener('click', (event: MouseEvent) => {
+        event.preventDefault();
+
+        onDrawerClose();
+        dispatch(collectionUiActions.setIsNew(true));
+        dispatch(collectionUiActions.initCollectionForm({}));
+        dispatch(commonUiActions.showDialog(DIALOGS.NEW_COLLECTION));
+      }, false);
+    }
```

Comment:
One thing to note from the outset: the collection list page will soon be getting the Create Collection button. If it's not difficult to move it to its own file, we should do that here

Reply:
Sure, I'll see about moving the logic to a library function